libres/lib/analysis/ies/ies_enkf.cpp

```diff
@@ -490,12 +446,13 @@ void ies_enkf_linalg_extract_active_W(const ies_enkf_data_type *data,
 *     Omega^T S^T = Y^T
 */
 void ies_linalg_solve_S(const matrix_type *W0, const matrix_type *Y,
-                        matrix_type *S, double rcond, FILE *log_fp, bool dbg) {
+                        matrix_type *S) {
     int ens_size = matrix_get_columns(W0);
     int nrobs = matrix_get_rows(S);
     int m_ens_size = std::min(ens_size - 1, 16);
     int m_nrobs = std::min(nrobs - 1, 7);
     double nsc = 1.0 / sqrt(ens_size - 1.0);
+    double rcond;
```

Comment:
This seems to remain a bit of a unused vestige now. Generally the old c programming style of declaring variables at the top of the scope is generally considered bad practice. Because it is unused (except to give matrix_dgesvx an output), its very confusing. So I suggest this variable is moved to where it is used:
```
double rcond;
matrix_dgesvx(Omega, YT, &rcond);
```
I think maybe `matrix_dgesvx` should accept `nullptr` to mean "Do not give me that output", which is a common practice. I am not sure whether lapack does that already, but I believe the underlying fortran code makes no such check.

Reply:
> I think maybe matrix_dgesvx should accept nullptr to mean "Do not give me that output"

That is a good idea.

> [...] but I believe the underlying fortran code makes no such check.

Fortran is call by reference and makes no such check.